src/modules/ens/sagas.ts

```diff
@@ -72,6 +78,28 @@ import { addWorldStatusToEachENS, getLandRedirectionHashes, isExternalName } fro
 const DEFAULT_ENS_PAGE_SIZE = 12
 
 export function* ensSaga(builderClient: BuilderClient, ensApi: ENSApi, worldsAPIContent: WorldsAPI) {
+  /** Get a provider for reading ENS data from Ethereum network.
+   * This allows ENS reads to work regardless of which network the user is connected to. */
+  function* getEthereumReadProvider(): Generator<any, ethers.providers.Web3Provider, any> {
+    const ethereumChainId: number = yield call(getChainIdByNetwork, Network.ETHEREUM)
+    const networkProvider: Provider = yield call(getNetworkProvider, ethereumChainId)
+    return new ethers.providers.Web3Provider(networkProvider)
+  }
+
+  /** Validate that the user's wallet is on Ethereum network for write operations. */
+  function* validateAndSwitchNetwork() {
+    const ethereumChainId: number = yield call(getChainIdByNetwork, Network.ETHEREUM)
+    const wallet: Wallet = yield getWallet()
+    if (wallet.chainId !== ethereumChainId) {
+      const signer: ethers.Signer = yield call(getSigner)
+      const signerNetwork: ethers.providers.Network = yield call([signer.provider as ethers.providers.Provider, 'getNetwork'])
+      if (signerNetwork.chainId !== ethereumChainId) {
+        yield put(switchNetworkRequest(ethereumChainId, signerNetwork.chainId))
+        yield take(SWITCH_NETWORK_SUCCESS)
+      }
```

Comment:
I think the `wallet.chainId !== ethereumChainId` validation works as the `switchNetworkRequest` updates the wallet state.

so the `wallet.chainId` and `signerNetwork.chainId` should be equal